common/src/verifier.rs

```diff
@@ -66,20 +98,24 @@ impl ImageVerificationEnv for &mut FirmwareImageVerificationEnv<'_, '_> {
         len: u32,
         digest_failure: CaliptraError,
     ) -> CaliptraResult<ImageDigest384> {
-        let mut digest = Array4x12::default();
-
-        if let Some(mut sha_acc_op) = self
-            .sha2_512_384_acc
-            .try_start_operation(ShaAccLockState::NotAcquired)?
-        {
-            sha_acc_op
-                .digest_384(len, offset, false, &mut digest)
-                .map_err(|_| digest_failure)?;
+        if self.image_in_mcu {
+            // For MCU case, use the existing sha384_digest function
+            self.sha384_digest(offset, len).map_err(|_| digest_failure)
         } else {
-            Err(CaliptraError::KAT_SHA2_512_384_ACC_DIGEST_START_OP_FAILURE)?;
-        };
-
-        Ok(digest.0)
+            let mut digest = Array4x12::default();
+
+            if let Some(mut sha_acc_op) = self
+                .sha2_512_384_acc
+                .try_start_operation(ShaAccLockState::NotAcquired)?
+            {
+                sha_acc_op
+                    .digest_384(len, offset, false, &mut digest)
+                    .map_err(|_| digest_failure)?;
+            } else {
+                Err(CaliptraError::KAT_SHA2_512_384_ACC_DIGEST_START_OP_FAILURE)?;
```

Comment:
Is this the right error code? It seems weird it is prefixed with `KAT_`. Same comment for SHA512 below.